handler/handler.go

```diff
@@ -47,29 +49,35 @@ func NewSpotHandler(
 		log:              log,
 		nodeName:         nodeName,
 		pollWaitInterval: pollWaitInterval,
+		gracePeriod:      30 * time.Second,
 	}
 }
 
 func (g *SpotHandler) Run(ctx context.Context) error {
 	t := time.NewTicker(g.pollWaitInterval)
 	defer t.Stop()
 
+	var once sync.Once
+	deadline := time.NewTimer(24 * 365 * time.Hour)
```

Comment:
why is this needed?

Reply:
this is a simple way to keep checking for preemption even if we received a termination signal, and when deadline actually occurs we exit the application